PathKit/PathKit.swift

```diff
@@ -90,94 +108,556 @@ public struct Path : Equatable, Hashable, CustomStringConvertible, StringLiteral
 
     /// Normalizes the path, this cleans up redundant ".." and ".", double slashes
     /// and resolves "~".
+    ///
+    /// - Returns: a new path made by removing extraneous path components from the underlying String
+    ///   representation.
+    ///
     public func normalize() -> Path {
         return Path((self.path as NSString).stringByStandardizingPath)
     }
+    
+    /// De-normalizes the path, by replacing the current user home directory with "~".
+    ///
+    /// - Returns: a new path made by removing extraneous path components from the underlying String
+    ///   representation.
+    ///
+    public func abbreviate() -> Path {
+        return Path((self.path as NSString).stringByAbbreviatingWithTildeInPath)
+    }
 
-    /// Returns whether a file or directory exists at a specified path
-    public func exists() -> Bool {
-        return NSFileManager().fileExistsAtPath(self.path)
+    /// Returns the path of the item pointed to by a symbolic link.
+    ///
+    /// - Returns: the path of directory or file to which the symbolic link refers
+    ///
+    public func symlinkDestination() throws -> Path {
+        let symlinkDestination = try Path.fileManager.destinationOfSymbolicLinkAtPath(path)
+        let symlinkPath = Path(symlinkDestination)
+        if symlinkPath.isRelative {
+            return self + ".." + symlinkPath
+        } else {
+            return symlinkPath
+        }
     }
+}
+
 
+// MARK: Path Components
+
+extension Path {
+    /// The last path component
+    ///
+    /// - Returns: the last path component
+    ///
+    public var lastComponent: String {
+        return (path as NSString).lastPathComponent
+    }
+    
+    /// The last path component without file extension
+    ///
+    /// - Note: This returns "." for "..".
+    ///
+    /// - Returns: the last path component without file extension
+    ///
+    public var lastComponentWithoutExtension: String {
+        return (lastComponent as NSString).stringByDeletingPathExtension
+    }
+    
+    /// Splits the string representation on the directory separator.
+    /// Absolute paths remain the leading slash as first component.
+    ///
+    /// - Returns: all path components
+    ///
+    public var components: [String] {
+        return (path as NSString).pathComponents
+    }
+    
+    /// The file extension behind the last dot of the last component.
+    ///
+    /// - Returns: the file extension
+    ///
+    public var `extension`: String {
```

Comment:
I think it's probably less surprising if it behaves similar to [the underlying Foundation API](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSString_Class/#//apple_ref/occ/instp/NSString/pathExtension), which returns an empty string instead. Otherwise I don't see arguments why one of both should be better or worse.